generate/generate_test.go

```diff
@@ -58,11 +59,96 @@ func (s stringer) String() string {
 	return string(s)
 }
 
+func TestGenerateConflictsBetweenGenerateTypes(t *testing.T) {
+	testCodeGeneration(t, assertHCLEquals, []testcase{
+		{
+			name: "stack with different generate blocks but same label",
+			layout: []string{
+				"s:stacks/stack",
+			},
+			configs: []hclconfig{
+				{
+					path: "/stacks",
+					add: generateHCL(
+						labels("repeated"),
+						content(
+							block("block",
+								str("data", "parent data"),
+							),
+						),
+					),
+				},
+				{
+					path: "/stacks/stack",
+					add: generateFile(
+						labels("repeated"),
+						str("content", "test"),
+					),
+				},
+			},
+			wantReport: generate.Report{
+				Failures: []generate.FailureResult{
+					{
+						Result: generate.Result{
+							StackPath: "/stacks/stack",
+						},
+						Error: errors.E(generate.ErrConflictingConfig),
+					},
+				},
+			},
+		},
+		{
+			name: "stack with block with same label as parent but different condition",
+			layout: []string{
+				"s:stacks/stack",
+			},
+			configs: []hclconfig{
+				{
+					path: "/stacks",
+					add: generateHCL(
+						labels("repeated"),
+						content(
+							block("block",
+								str("data", "parent data"),
+							),
+						),
+						boolean("condition", false),
+					),
+				},
+				{
+					path: "/stacks/stack",
+					add: generateFile(
+						labels("repeated"),
+						str("content", "test"),
+					),
+				},
+			},
+			want: []generatedFile{
+				{
+					stack: "/stacks/stack",
+					files: map[string]fmt.Stringer{
+						"repeated": stringer("test"),
+					},
+				},
+			},
+			wantReport: generate.Report{
+				Successes: []generate.Result{
+					{
+						StackPath: "/stacks/stack",
+						Created:   []string{"repeated"},
+					},
+				},
+			},
+		},
+	})
```

Comment:
Not fundamental, but maybe add a test where we have generate_hcl + generate_file on the same level (like the stack) but it works because of the condition (checking it works when it is unrelated to hierarchy).

Reply:
test added. Thanks